content/2022/prepoznavanje-govora.md

```diff
@@ -2,3 +2,237 @@
+
+Osvrt na rad ogleda se u setu metoda koje su pokrivene u referentnim radovima. Uloga ovih metoda može se podeliti u nekoliko kategorija: 
+
+1. Izvlačenje karakteristika iz zvuka pomoću kepstralnih koeficijenata Mel skale (MFCC) 
+
+2. Klasifikatori, kojima su prosleđene MFCC karakteristike: Logistička regresija, Random Forest, SVM, XGBoost;
+
+3. Konvolucione neuronske mreže (CNN) koje inkomponuju proces ekstrakcije karakteristika iz signala, kao i proces klasifikacije 
+
+### Metode
+
+Rešenje datog problema prepoznavanja govora svodi se na izradu spektrograma i obradu istih.
+
+#### Spektrogrami
+
+Spektrogrami su vizuelne reprezentacije jačine signala. Mogu se posmatrati kao dvodimenzionalni grafici gde se može uočiti i treća dimenzija preko boja svakog dela spektrograma. Vremenska osa se gleda sa leve na desnu stranu po horizontalnoj osi. Vertikalna osa predstavlja frekvencijske komponente prisutne u signalu, dok boja označava jačinu svake od tih komponenti. U logaritamskoj je skali kako bi se prilagodila ljudskom uhu koje čuje po istom principu, što je dalje objašnjeno u samom radu.
+
+Spektrogram služi za prikazivanje aplitude svake frekvencijske komponente signala u vremenskom intervalu. Intervali su mali da bi se moglo pretpostaviti da se amplitude frekvencijskih komponenti ne menjaju.
+
+Boja na grafiku predstavlja amplitudu signala u određenom vremenskom trenutku. Plava boja na spektrogramu predstavlja niske amplitude, dok crvena boja predstavlja visoke amplitude.
+
+![spektrogram](static\images\2.png)
+
+#### Metode obrade spektrograma
+
+##### 1. Logistička regresija
+
+Logistička regresija je metoda klasifikacije koja se može primeniti i koristiti svuda gde imamo promenljive koje se mogu kategorisati. Za razliku od linearne regresije, vrednosti njenih rezultata su ograničene između 0 i 1. 
+
+U slučaju binarne klasifikacije, ova metoda umesto linearne koristi sigmoidnu funkciju. U slučaju više klasa, koristi se softmax funkcija. Sigmoidna i softmax funckija prikazane su na slici:
+
+![Sigmoid](static\images\Sigmoid.svg)
+
+Binarna logistička regresija kao izlaz daje vrednosti 0 ili 1, zavisno od toga da li posmatrana promenljiva pripada nekoj klasi ili ne. U slučaju kada imamo više od dve klase, koristi se multinomijalna logistička regresija (Softmax Regression). Na sledećoj slici prikazana je Softmax funkcija.
+
+![Softmax](static\images\Softmax.svg)
+
+Kriterijumska funkcija ove metode je logaritamska kako bi se postiglo da gradient descent nađe globalni, a ne samo lokalni minimum funkcije.
+
+$$\begin{equation}
+\sigma(\vec{z})_i=e^{z_i}*(\sum_{j=1}^{K}e^{z_j})^{-1} 
```

Comment:
Nisam sigurna čemu ova strelica iznad vrednosti z. Ako hoćete da naglasite da je vektor, predstavite kao boldovan tekst.

Reply:
Takođe, ako je to z vektor, onda ove dve strane ne čine jednakost. S desne strane je rezultat skalar.

Reply:
Imamo tu problem oko tog zapisa, neće da se lepo napiše ma kako stavili, pokazali smo vam već za taj slučaj, a ima i još nekoliko mesta